app/helpers/audit_helper.rb

```diff
@@ -87,7 +87,23 @@ def activity(type_of_activity, args = nil)
     end
   end
 
-  def define_user(audit)
-    audit.activity_type.include?("received") ? "Applicant / Agent via #{audit.api_user.name}" : audit.api_user.name
+  def define_audit_user(audit)
+    if audit.api_user.present?
+      audit.activity_type.include?("received") ? "Applicant / Agent via #{audit.api_user.name}" : audit.api_user.name
+    elsif audit.user.present?
+      audit.user.name
+    else
+      "User deleted"
+    end
+  end
+
+  def audit_type(audit)
+    if audit.activity_type.match("/*_validation_request_cancelled")
+      render "audits/types/validation_request_cancelled", locals: { item: audit }
+    elsif audit.activity_type.include?("request") || audit.activity_type.include?("document_received_at_changed")
+      render "audits/types/#{audit.activity_type}", locals: { item: audit }
+    else
+      render "audits/types/generic_audit_entry", locals: { item: audit }
```

Comment:
sorry I was unclear but can you make this function solely compute the template name and keep the rendering in the view? I've made you a patch, if you're happy with it :gift: 

```patch
From f367cfa86d03e9d3f3402a0bbecbbe159eb11485 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Maniaci?=
 <stephane.maniaci@unboxedconsulting.com>
Date: Fri, 10 Dec 2021 11:06:44 +0000
Subject: [PATCH] audit: make the helper do less

---
 app/helpers/audit_helper.rb                       | 8 ++++----
 app/views/shared/_supporting_information.html.erb | 6 +++---
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/app/helpers/audit_helper.rb b/app/helpers/audit_helper.rb
index 35a19acb..e2031a37 100644
--- a/app/helpers/audit_helper.rb
+++ b/app/helpers/audit_helper.rb
@@ -97,13 +97,13 @@ module AuditHelper
     end
   end
 
-  def audit_type(audit)
+  def audit_entry_template(audit)
     if audit.activity_type.match("/*_validation_request_cancelled")
-      render "audits/types/validation_request_cancelled", locals: { item: audit }
+      "validation_request_cancelled"
     elsif audit.activity_type.include?("request") || audit.activity_type.include?("document_received_at_changed")
-      render "audits/types/#{audit.activity_type}", locals: { item: audit }
+      audit.activity_type
     else
-      render "audits/types/generic_audit_entry", locals: { item: audit }
+      "generic_audit_entry"
     end
   end
 end
diff --git a/app/views/shared/_supporting_information.html.erb b/app/views/shared/_supporting_information.html.erb
index ec873b31..f67f6508 100644
--- a/app/views/shared/_supporting_information.html.erb
+++ b/app/views/shared/_supporting_information.html.erb
@@ -124,13 +124,13 @@
     <div class="govuk-accordion__section-content" aria-labelledby="audit-log">
       <% if @last_audit.present? %>
         <p class="govuk-body">
-          <%= audit_type(@last_audit) %>
-  
+          <%= render "audits/types/#{audit_entry_template(@last_audit)}", locals: { item: @last_audit } %>
+
           <p class="govuk-body">
             <%= define_audit_user(@last_audit) %>
           </p>
           <p class="govuk-body">
-            <%= @last_audit.created_at.to_date %> at <%= @last_audit.created_at.strftime("%H:%M") %> 
+            <%= @last_audit.created_at.to_date %> at <%= @last_audit.created_at.strftime("%H:%M") %>
           </p>
         </p>
       <% end %>
-- 
2.33.1

```

Reply:
ahh awesome I hadn't understood that's how you were intending it, will put in that logic thanks